### PR TITLE
fix multiple match

### PR DESCRIPTION
**Pull request by czpmango**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
close https://github.com/vesoft-inc/nebula/issues/4069
## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [x] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/graph/planner/match/MatchPlanner.cpp [hunk 1]
```diff
@@ -62,89 +56,78 @@ StatusOr<SubPlan> MatchPlanner::genPlan(CypherClauseContextBase* clauseCtx) {
...
 
-  queryPlan = SegmentsConnector::addInput(boundaryPlan.value(), queryPlan);
   return Status::OK();
 }
 
+Status MatchPlanner::genQueryPartPlan(QueryContext* qctx,
+                                      SubPlan& queryPlan,
+                                      const QueryPart& queryPart) {
+  // generate plan for matchs
+  for (auto& match : queryPart.matchs) {
```
Comment (Aiee) [inline src/graph/planner/match/MatchPlanner.cpp:103]: matches

File: src/graph/planner/match/MatchPlanner.cpp [hunk 2]
```diff
@@ -62,89 +56,84 @@ StatusOr<SubPlan> MatchPlanner::genPlan(CypherClauseContextBase* clauseCtx) {
...
-  for (auto& alias : firstMatch->aliasesGenerated) {
-    if (aliasesAvailable.find(alias.first) != aliasesAvailable.end()) {
-      intersectedAliases.insert(alias.first);
-    }
+  // generate plan for boundary
+  auto boundaryPlanStatus = genPlan(queryPart.boundary.get());
+  NG_RETURN_IF_ERROR(boundaryPlanStatus);
+  auto boundaryPlan = std::move(boundaryPlanStatus).value();
+  if (queryPlan.root == nullptr) {
+    queryPlan = boundaryPlan;
```
Comment (jievince) [inline src/graph/planner/match/MatchPlanner.cpp:119]: Which case?

Comment (jievince) [reply src/graph/planner/match/MatchPlanner.cpp:119]: Got it.

**Review by jievince — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Comment by czpmango** [comment]
Refer to https://core.ac.uk/download/pdf/148787624.pdf

**Pull request merged**